# ARC Test on Llama 2

In [1]:
import os
import json
import replicate
import re
import pandas as pd
from tqdm import tqdm
from transformers import LlamaTokenizer

## Connection to the API

In [2]:
os.environ['REPLICATE_API_TOKEN'] = 'r8_TPbThLzct109l1IBPqPDWoLB1GdVPPD4f3CaT'

## Number of Tokens

Since the number of tokens which can be fed to Llama 2 is limited to 4096, we need to count the number of tokens present in each prompt we are generating before sending them. System prompt is always sent to the model together with the main prompt. Let's calculate the number of tokens in system prompt:

In [3]:
system_prompt = """
You will be given a list of input-output pairs labeled "Case 0" "Case 1" and so on.

Each input and output is a grid of numbers representing a visual grid. There is a SINGLE rule that transforms each input
grid to the corresponding output grid.

The pattern may involve counting or sorting objects (e.g. sorting by size) comparing numbers (e.g. which shape or symbol
appears the most? Which is the largest object? Which objects are the same size?) or repeating a pattern for a fixed number
of time.

There are other concepts that may be relevant.

- Lines rectangular shapes

- Symmetries rotations translations.

- Shape upscaling or downscaling elastic distortions.

- Containing / being contained / being inside or outside of a perimeter.
- Drawing lines connecting points orthogonal projections.

- Copying repeating objects.

You should treat cells with 0 as empty cells (backgrounds) .

Please generate the Output grid that corresponds to the last given Input grid using the transformation rule you induced
from the previous input-output pairs. Just give the output with no further words.
"""

We will use the below tokenizer througout the code:

In [4]:
# Initialize the tokenizer
tokenizer = LlamaTokenizer.from_pretrained("hf-internal-testing/llama-tokenizer")

You are using the default legacy behaviour of the <class 'transformers.models.llama.tokenization_llama.LlamaTokenizer'>. This is expected, and simply means that the `legacy` (previous) behavior will be used so nothing changes for you. If you want to use the new behaviour, set `legacy=False`. This should only be set if you understand what it means, and thoroughly read the reason why this was added as explained in https://github.com/huggingface/transformers/pull/24565


In [5]:
# Encode the system prompt
encoded_prompt = tokenizer.encode(system_prompt)

# Calculate the number of tokens
num_tokens = len(encoded_prompt)

print(f"The number of tokens in the system prompt is {num_tokens}.")

The number of tokens in the system prompt is 269.


Therefore, for the main prompt we can tolerate 4096 - 269 = 3827 ~ 3820 tokens.

A couple of useful functions for the rest of the code:

In [6]:
def keep_numbers_only(input_string):         # this is a function  to create a serial version of the results
    # Use regular expression to remove non-numeric characters
    numbers_only = re.sub(r'[^0-9]', '', input_string)
    return numbers_only

In [7]:
def new_content(result , file_content ):

  new_content = re.sub(r"[^0-9\s\[\]\n]", "", result).replace("\n", "")

  file_content += new_content + '\n [INST] The last output you produced is wrong, try again by predicting another output for the last input given. [/INST]' + '\n output:'

  return file_content

# Run The Full Test on The Model

In [20]:
Comparison = []

folder_path = 'New_Prompts/'

# Get list of files with their sizes
files_with_sizes = [(f, os.path.getsize(os.path.join(folder_path, f))) for f in os.listdir(folder_path)]

# Filter files to keep only those smaller than 5 KB (5 * 1024 bytes)
filtered_files = [(name, size) for name, size in files_with_sizes if size < 5 * 1024]

# Sort filtered files based on their sizes in descending order
sorted_files = sorted(filtered_files, key=lambda x: x[1], reverse=True)

# Extract only the filenames from the sorted list
sorted_filenames = [f[0] for f in sorted_files]

print(len(sorted_filenames))

480


In [25]:
processed_files = set()

# Collect the names of already processed files
for entry in Comparison:
    processed_files.add(entry[0])

In [26]:
processed_files

{'AboveBelow10_0.json',
 'AboveBelow10_1.json',
 'AboveBelow10_2.json',
 'AboveBelow1_0.json',
 'AboveBelow1_1.json',
 'AboveBelow1_2.json',
 'AboveBelow2_0.json',
 'AboveBelow2_1.json',
 'AboveBelow2_2.json',
 'AboveBelow7_0.json',
 'AboveBelow7_1.json',
 'AboveBelow7_2.json',
 'AboveBelow8_0.json',
 'AboveBelow8_1.json',
 'AboveBelow8_2.json',
 'AboveBelow9_0.json',
 'AboveBelow9_1.json',
 'AboveBelow9_2.json',
 'Center1_2.json',
 'Center3_0.json',
 'Center3_1.json',
 'Center3_2.json',
 'Center5_0.json',
 'Center5_1.json',
 'Center5_2.json',
 'Center6_0.json',
 'Center6_1.json',
 'Center6_2.json',
 'Center8_0.json',
 'Center8_1.json',
 'Center8_2.json',
 'Center9_1.json',
 'CleanUp1_0.json',
 'CleanUp3_0.json',
 'CleanUp3_1.json',
 'CleanUp3_2.json',
 'CleanUp4_0.json',
 'CleanUp4_1.json',
 'CleanUp4_2.json',
 'CleanUp5_0.json',
 'CleanUp5_1.json',
 'CleanUp5_2.json',
 'CleanUp6_0.json',
 'CleanUp6_1.json',
 'CleanUp6_2.json',
 'CleanUp7_0.json',
 'CleanUp7_1.json',
 'CleanUp7_2.json

In [27]:
for f in tqdm(sorted_filenames, desc="Processing files", unit="file"):
    
    # Check if the file has already been processed
    if f in processed_files:
        print(f"{f} already processed. Skipping...")
        continue
    
    ignore_file = False
    
    prompt_path = os.path.join('New_Prompts/', f)
    correctanswer_path = os.path.join('New_Correct_Answers/', f)

    with open(prompt_path, 'r') as file:
        prompt_content = file.read()

    with open(correctanswer_path, 'r') as file:
        answer_content = file.read()

    prompt = ' """ ' + prompt_content + ' """ '
    
    trial = 0

    for i in range(3):
        encoded_prompt = tokenizer.encode(prompt)
        
        if len(encoded_prompt) < 3810:
            trial += 1

            output_generator = replicate.run(
                "meta/llama-2-13b-chat:f4e2de70d66816a838a89eeeb621910adffb0dd0baba3976c96980970978018d",
                input={
                    "prompt": prompt,
                    "system_prompt": system_prompt,
                    "max_new_tokens": 2048,
                    "temperature": 0.5
                }
            )

            output_list = list(output_generator)
            output_string = " ".join(output_list)
            result = output_string

            model_answer = keep_numbers_only(result)
            correct_answer = keep_numbers_only(answer_content)

            if model_answer == correct_answer:
                break

            prompt_content = new_content(result, prompt_content)
            prompt = ' """ ' + prompt_content + ' """ '
        else:
            if trial == 0:
                ignore_file = True
                break
            else:
                break
    
    if ignore_file:
        Comparison.append([f, "NA", "NA", "NA", "NA", trial])
    else:
        Comparison.append([f, model_answer, correct_answer, model_answer == correct_answer,
                           len(model_answer) == len(correct_answer), trial])

    processed_files.add(f)
    print(Comparison[-1])


Processing files:   0%|          | 0/480 [00:00<?, ?file/s]

ExtendToBoundary2_1.json already processed. Skipping...
AboveBelow2_2.json already processed. Skipping...
ExtendToBoundary2_2.json already processed. Skipping...
InsideOutside4_0.json already processed. Skipping...
InsideOutside4_1.json already processed. Skipping...
InsideOutside4_2.json already processed. Skipping...
ExtendToBoundary2_0.json already processed. Skipping...
AboveBelow2_1.json already processed. Skipping...
AboveBelow2_0.json already processed. Skipping...
InsideOutside5_2.json already processed. Skipping...
InsideOutside5_1.json already processed. Skipping...
HorizontalVertical8_1.json already processed. Skipping...
InsideOutside5_0.json already processed. Skipping...
ExtendToBoundary9_0.json already processed. Skipping...
HorizontalVertical8_0.json already processed. Skipping...
ExtendToBoundary9_1.json already processed. Skipping...
HorizontalVertical4_2.json already processed. Skipping...
Count2_2.json already processed. Skipping...
HorizontalVertical8_2.json alread

Processing files:  52%|█████▏    | 249/480 [00:07<00:06, 35.51file/s]

['ExtractObjects4_2.json', '333', '33333333333333333333', False, False, 3]


Processing files:  52%|█████▏    | 250/480 [00:38<00:48,  4.77file/s]

['Order6_0.json', '0000000000400000000033333330003333333000333333300033333330003333333000333333300077000000007700000000666660000066666000006666600000666660000012345', '00000000004000000000770000000077000000006666600000666660000066666000006666600000333333300033333330003333333000333333300033333330003333333000', False, False, 3]


Processing files:  52%|█████▏    | 251/480 [00:59<01:26,  2.65file/s]

['FilledNotFilled6_2.json', '0000000055555550', '00000000000000000001111110000000111111100000111111111100001111111111000011111111110000001111111100000111111111000011111111110000111111111100000000000000000000000000000000007777777770000070000000700000700000007000007000000070000070000000700000777777777000000000000000', False, False, 3]


Processing files:  52%|█████▎    | 252/480 [01:43<03:19,  1.14file/s]

['CleanUp8_2.json', '77000000000011000000000044000000003300000000005500000000006600000000770000000088', '111100000000111100000000111100000000111100000000000044440000000044440000000044440000000044440000000000008888000000008888000000008888000000008888', False, False, 3]


Processing files:  53%|█████▎    | 253/480 [01:52<03:48,  1.01s/file]

['AboveBelow3_0.json', '5555555', '000000000000000000000000000001111100005555500005555500005555500005555500005555500000000000000000000001111100', False, False, 3]


Processing files:  53%|█████▎    | 254/480 [02:14<05:33,  1.47s/file]

['TopBottom2D10_1.json', '00000000000001110000000111000000022200000002220000000333300000003333000000033330000004444440000444444000044444400', '0000000000004444440000444444000044444400000333300000033330000003333000000333300000002200000000220000', False, False, 3]


Processing files:  53%|█████▎    | 255/480 [02:37<08:00,  2.13s/file]

['AboveBelow6_1.json', '00000000000000000000000000000000000000088888888888888888888822222222288888888888888888888880000000000000000000000000000000000000000', '0000000000000000000000000000000000000008888888888888888888882222222228888222222222888822222000000000000000000000000000000000000000', False, False, 3]


Processing files:  53%|█████▎    | 256/480 [02:44<08:43,  2.34s/file]

['TopBottom2D3_2.json', '555555555', '000000000000000000000000002222222200000001100000000000000000', False, False, 3]


Processing files:  54%|█████▎    | 257/480 [03:00<11:37,  3.13s/file]

['AboveBelow3_2.json', '5555555000', '0000000000000000000000055550000005555000000555500000055550000005555000000000000000000000000000000000', False, False, 3]


Processing files:  54%|█████▍    | 258/480 [03:13<14:11,  3.84s/file]

['ExtractObjects3_1.json', '80008', '555555500005500005500005500005555555', False, False, 3]


Processing files:  54%|█████▍    | 259/480 [03:25<17:22,  4.71s/file]

['FilledNotFilled5_0.json', '000444000004000400040000040400000004444444444000000000', '000444000004444400044444440444444444444444444000000000', False, True, 3]


Processing files:  54%|█████▍    | 260/480 [03:32<17:56,  4.89s/file]

['SameDifferent5_0.json', '6', '8', False, True, 3]


Processing files:  54%|█████▍    | 261/480 [03:49<24:44,  6.78s/file]

['Order4_2.json', '888033307770555888033307770555888033307770555888033307770555888033307770555888033307770555888033307770555', '000000000000000000000000000000000000000000000555088803330777555088803330777555088803330777', False, False, 3]


Processing files:  55%|█████▍    | 262/480 [04:02<28:23,  7.81s/file]

['Center9_2.json', '5555555000000000000002000000555555500000000000000', '5555555000000020000000000000555555500000000000000', False, True, 3]


Processing files:  55%|█████▍    | 263/480 [05:00<1:04:23, 17.81s/file]

['Center1_1.json', '0666660000006', '000000000000000000000000000000066600006666666066600006000006066600006000006000000006006006000000006000006066666006000006060006006666666060606000000000060006000000000066666000000000000000000000000000000000000000000000000000000', False, False, 3]


Processing files:  55%|█████▌    | 264/480 [06:59<2:25:21, 40.38s/file]

['CompleteShape4_1.json', '88888888', '000222222000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002222220', False, False, 3]


Processing files:  55%|█████▌    | 265/480 [07:29<2:15:47, 37.90s/file]

['CompleteShape8_2.json', '0000000000000222000000000050500000000000000000000000006660000000666000000066600000000000000000000000', '000000000000000000000000000000000000000000020000000020200000020002000000202005000002005050000005000500000050500000000500', False, False, 3]


Processing files:  55%|█████▌    | 266/480 [07:49<1:59:04, 33.39s/file]

['AboveBelow4_0.json', '009333000000000000093330666666666666000000000000000000000000000000333333000000000000000000000000000000000000', '000000000000000133330000000000000000000000000000000000000133111111111111000000000000033333100000000000000331', False, True, 3]


Processing files:  56%|█████▌    | 267/480 [07:59<1:35:59, 27.04s/file]

['CleanUp1_2.json', '4646464684', '464646464464646464464646464464646464464646464464646464464646464464646464464646464', False, False, 3]


Processing files:  56%|█████▌    | 268/480 [08:06<1:15:26, 21.35s/file]

['FilledNotFilled1_1.json', '33333333', '8', False, False, 3]


Processing files:  56%|█████▌    | 269/480 [08:12<59:44, 16.99s/file]  

['Order2_0.json', '777777777', '000000000000000000000000000000000000000000000000000700000000007700000000007700000000077700000000077700000000777700000000', False, False, 3]


Processing files:  56%|█████▋    | 270/480 [08:19<49:23, 14.11s/file]

['AboveBelow3_1.json', '000011110', '000000000000011100004444400004000400004000400004000400004444400000000000000111100', False, False, 3]


Processing files:  56%|█████▋    | 271/480 [08:24<39:50, 11.44s/file]

['ExtractObjects4_1.json', '555000', '555555', False, True, 3]


Processing files:  57%|█████▋    | 272/480 [08:33<37:19, 10.77s/file]

['Order4_0.json', '88803330555', '000000000703010206703010206703010206703010206703010206703010206703010206', False, False, 3]


Processing files:  57%|█████▋    | 273/480 [08:38<31:28,  9.12s/file]

['Center9_0.json', '030803', '030003030003030803030003030003030003', False, False, 3]


Processing files:  57%|█████▋    | 274/480 [08:44<27:46,  8.09s/file]

['TopBottom3D5_1.json', '1111111', '444444444444', False, False, 3]


Processing files:  57%|█████▋    | 275/480 [09:04<39:49, 11.66s/file]

['TopBottom3D1_1.json', '00200020000020002000002000200008888888880088888888800888888888008888888880088888888800020002000', '0020002000002000200000200020000828882880082888288008288828800828882880082888288008288828800020002000', False, False, 3]


Processing files:  57%|█████▊    | 276/480 [09:30<54:11, 15.94s/file]

['TopBottom3D1_2.json', '044400444403333303333333330333333333033333333303333044400444400000034443000003444300000344430000034443', '0444000000344430000034443000003444300000344430000004440044400000033333000003333300000333330000033333', False, False, 3]


Processing files:  58%|█████▊    | 277/480 [09:41<49:12, 14.54s/file]

['TopBottom3D3_2.json', '9999999', '9000000000090000000000900000000009000000000090000000000900000000009000000000090000000000900000000009', False, False, 3]


Processing files:  58%|█████▊    | 278/480 [09:46<39:23, 11.70s/file]

['FilledNotFilled1_2.json', '9', '1', False, True, 3]


Processing files:  58%|█████▊    | 279/480 [09:50<31:44,  9.48s/file]

['ExtendToBoundary5_0.json', '70000', '700000007070000070007000700000707000000070000000707000007000700070000070700000007', False, False, 3]


Processing files:  58%|█████▊    | 280/480 [09:55<27:04,  8.12s/file]

['ExtendToBoundary5_1.json', '70000', '000000007000000070700000700070007000007070000000700000007070000070007000700000700', False, False, 3]


Processing files:  59%|█████▊    | 281/480 [10:15<38:48, 11.70s/file]

['CleanUp8_0.json', '3300030009330300000000220000600022009000000065006004405500000000007700060000476006004000660000000066', '3300000000330000000000220000000022000000000055000000005500000000007700000000770000000000660000000066', False, True, 3]


Processing files:  59%|█████▉    | 282/480 [10:20<31:40,  9.60s/file]

['CompleteShape3_0.json', '777777777', '0000000000000000000007777777700700000070070000007007000000700700000070070000007007000000700777777770', False, False, 3]


Processing files:  59%|█████▉    | 283/480 [10:33<35:15, 10.74s/file]

['CompleteShape4_0.json', '0333333333033333333333333333333333333333333333333333333333333333333333333333333333333333333', '0000000000000000000003000000300300000030030000003003000000300000000000000000000000000000000000000000', False, False, 3]


Processing files:  59%|█████▉    | 284/480 [10:49<39:28, 12.08s/file]

['CompleteShape8_0.json', '00000000000008880000', '0000000000000000000007070000000700000000077700000000000000000000000000000080800000008000000000888000', False, False, 3]


Processing files:  59%|█████▉    | 285/480 [11:02<40:49, 12.56s/file]

['CompleteShape8_1.json', '00000000000008880000', '0000000000000000000000300000000303000000003000008000000008080090000080090900000000900000000000000000', False, False, 3]


Processing files:  60%|█████▉    | 286/480 [11:23<48:42, 15.07s/file]

['HorizontalVertical10_0.json', '999999999900000000000000000000000000000000000000000000000000000000000033333000004444400000555550000011111000000000000000', '0000000008000000000800000000080000000008000000000833333000084444400008555550000811111000080000000008', False, False, 3]


Processing files:  60%|█████▉    | 287/480 [11:46<55:52, 17.37s/file]

['HorizontalVertical10_1.json', '0000000009', '8888888888000000000000000000000000000000000000000000023000000002300000000230000000000000000000000000', False, False, 3]


Processing files:  60%|██████    | 288/480 [12:21<1:12:30, 22.66s/file]

['HorizontalVertical10_2.json', '8888888888000000000000000000000078910000007891000000789100000078910000000000000000000000000000000000', '0000000008000000000800000000080000000008000000000800000000080044444448006666666800000000080000000008', False, True, 3]


Processing files:  60%|██████    | 289/480 [12:40<1:08:26, 21.50s/file]

['MoveToBoundary3_0.json', '0000000000000000000000000000000222000000020200000002220000000000000000000000000000000000000000000000', '0000000000000000000000000000000000000222000000020200000002220000000000000000000000000000000000000000', False, True, 3]


Processing files:  60%|██████    | 290/480 [12:54<1:01:32, 19.44s/file]

['MoveToBoundary3_1.json', '6660000000', '0000000000000000000000000000000000000222000000000200000002220000000000000000000000000000000000000000', False, False, 3]


Processing files:  61%|██████    | 291/480 [13:00<47:59, 15.23s/file]  

['MoveToBoundary3_2.json', '222222220', '0000000000000000000000222222220000000000000000000000000000000000022222000000000000000000000000000000', False, False, 3]


Processing files:  61%|██████    | 292/480 [13:10<43:08, 13.77s/file]

['MoveToBoundary4_0.json', '6666000000', '0000000000000000000000000000000060000000666000000000600000006060000000000000000000000000000000000000', False, False, 3]


Processing files:  61%|██████    | 293/480 [13:16<35:12, 11.30s/file]

['MoveToBoundary4_1.json', '666666666', '0000000000666600000000060000000006000000000600000000060000000666000000000000000000000000000000000000', False, False, 3]


Processing files:  61%|██████▏   | 294/480 [13:22<30:26,  9.82s/file]

['MoveToBoundary4_2.json', '666666666', '0000000000000000000000000000006666666660000000000060000000006000000000600000000060000000006000000000', False, False, 3]


Processing files:  61%|██████▏   | 295/480 [13:40<37:54, 12.30s/file]

['MoveToBoundary5_0.json', '000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000', '0004000000000400000000040000000004000000004440000000000000000000000000000000000000000000000000000000', False, False, 3]


Processing files:  62%|██████▏   | 296/480 [15:39<2:15:39, 44.24s/file]

['MoveToBoundary5_1.json', '000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000', '0000000000000000000000000200000000020000000002000022222200000000020000000002000000000200000000000000', False, False, 3]


Processing files:  62%|██████▏   | 297/480 [15:54<1:48:11, 35.48s/file]

['MoveToBoundary5_2.json', '00000000000088880000008000000000000000000000888800000000080000000008000000000000', '0000000000000000800000000088880000008000000000000000080000008888000000000800000000000000000000000000', False, False, 3]


Processing files:  62%|██████▏   | 298/480 [16:11<1:30:49, 29.94s/file]

['SameDifferent4_0.json', '000000333300001000000000000000000000000000000000000000000000077777700000000000000000000000', '0000000000000010000000010000000010000500010000500010000500000000500000000000000000000000000000000000', False, False, 3]


Processing files:  62%|██████▏   | 299/480 [16:24<1:15:25, 25.00s/file]

['SameDifferent4_1.json', '0000000000002000000000200000000000000000000070000000007000000000700000000070000000000000000000000000', '0000000000002000000000200000000000000000000070000000007000000000700000000070000000000000000000000000', True, True, 2]


Processing files:  62%|██████▎   | 300/480 [16:34<1:01:08, 20.38s/file]

['SameDifferent4_2.json', '3000400009', '3000400000300040000030004000003000000000300000000030000000003000000000300000000030000000003000000000', False, False, 3]


Processing files:  63%|██████▎   | 301/480 [16:55<1:01:49, 20.72s/file]

['Order2_1.json', '0111000000011100000000000111100000000000000000000000100000000010000000000000100000000000000000000000', '0000000000000000000000000000000000000000000000000000000000000001000000000100000001110000001111000000', False, True, 3]


Processing files:  63%|██████▎   | 302/480 [17:00<47:12, 15.92s/file]  

['TopBottom2D10_2.json', '1111111', '00000000000000000000000000000044400004440000110000011000', False, False, 3]


Processing files:  63%|██████▎   | 303/480 [17:14<45:32, 15.44s/file]

['TopBottom2D10_0.json', '00110000110000220000220000660000660000770000770000000000111100', '007700007700002200002200006600006600001100001100000000', False, False, 3]


Processing files:  63%|██████▎   | 304/480 [17:26<41:48, 14.25s/file]

['Order2_2.json', '00000055000', '000000000000000000000000000000000000000000000000500000000555000000055550000005555500000055555000000', False, False, 3]


Processing files:  64%|██████▎   | 305/480 [17:31<33:48, 11.59s/file]

['Count3_0.json', '3333', '33333', False, False, 3]


Processing files:  64%|██████▍   | 306/480 [17:33<25:07,  8.66s/file]

['Count4_0.json', '0000', '0000', True, True, 1]


Processing files:  64%|██████▍   | 307/480 [18:00<40:28, 14.04s/file]

['MoveToBoundary1_1.json', '000111020000110020000100020000110020000111020000110020000100020000110020000111020000110020000100020000110020000111020', '000011120000011020000010020000011020000011120000011020000010020000011020000011120', False, False, 3]


Processing files:  64%|██████▍   | 308/480 [18:10<37:16, 13.00s/file]

['MoveToBoundary1_2.json', '2222222', '000000000000000000000000000000000000000000000010001000101010101000100010222222222', False, False, 3]


Processing files:  64%|██████▍   | 309/480 [18:35<47:11, 16.56s/file]

['AboveBelow6_0.json', '00000000000000000000000033333302203333330330222222033000000003300000000330000000', '00000000000000000000000033333302203333330220222222022000000002200000000220000000', False, True, 3]


Processing files:  65%|██████▍   | 310/480 [18:48<43:30, 15.35s/file]

['TopBottom2D7_1.json', '', '4444', False, False, 3]


Processing files:  65%|██████▍   | 311/480 [18:53<34:35, 12.28s/file]

['TopBottom2D7_2.json', '3333333', '777777777777777', False, False, 3]


Processing files:  65%|██████▌   | 312/480 [18:59<29:10, 10.42s/file]

['CleanUp2_0.json', '1', '111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111', False, False, 3]


Processing files:  65%|██████▌   | 313/480 [19:15<33:32, 12.05s/file]

['Copy7_1.json', '393300444400393000444003900004440030000000400000000000000000000000000000000000', '000000000000300000000400390000009400393000049400393300449400393000049400390000009400300000000400000000000000000000000000000000000000', False, False, 3]


Processing files:  65%|██████▌   | 314/480 [19:20<27:35,  9.97s/file]

['ExtractObjects3_2.json', '9999999', '999999999999900000000009900000000009900000000009900000000009900000000009900000000009900000000009900000000009900000000009900000000009999999999999', False, False, 3]


Processing files:  66%|██████▌   | 315/480 [19:29<26:56,  9.80s/file]

['ExtractObjects9_1.json', '8888333188888888', '10001111111101101110', False, False, 3]


Processing files:  66%|██████▌   | 316/480 [19:37<24:44,  9.05s/file]

['TopBottom3D5_0.json', '888666366', '4444044404444444', False, False, 3]


Processing files:  66%|██████▌   | 317/480 [19:57<33:48, 12.45s/file]

['CleanUp1_1.json', '88888899999988882899999988888899999988888899999988882899999988888899999901342341117223411821939', '888888999999888888999999888888999999', False, False, 3]


Processing files:  66%|██████▋   | 318/480 [20:02<27:50, 10.31s/file]

['ExtractObjects1_0.json', '8848', '88888880008000000800000080000008000', False, False, 3]


Processing files:  66%|██████▋   | 319/480 [20:10<25:45,  9.60s/file]

['ExtractObjects1_1.json', '333', '2222220002200022000222222', False, False, 3]


Processing files:  67%|██████▋   | 320/480 [20:16<22:20,  8.38s/file]

['ExtractObjects1_2.json', '773', '7777777', False, False, 3]


Processing files:  67%|██████▋   | 321/480 [20:25<22:56,  8.66s/file]

['AboveBelow4_1.json', '0033300000', '000034000000000000003333333333000000000000004444430000000000', False, False, 3]


Processing files:  67%|██████▋   | 322/480 [20:39<27:10, 10.32s/file]

['AboveBelow6_2.json', '000000000000000000000220002111000111100011110001111000111100011', '000000000000000000000220002122000212200021220002122000212200021', False, True, 3]


Processing files:  67%|██████▋   | 323/480 [20:54<30:08, 11.52s/file]

['TopBottom3D1_0.json', '000500000888888008888880055555500000000000000000000000000', '0000000000050000088588000885880000050000000000000000000000000000', False, False, 3]


Processing files:  68%|██████▊   | 324/480 [21:11<34:43, 13.36s/file]

['TopBottom3D3_0.json', '0000000000050000088588000885880000050000000000000000000000000000', '0000000000000000088888000888880000000000000000000000000000000000', False, True, 3]


Processing files:  68%|██████▊   | 325/480 [21:30<38:33, 14.93s/file]

['AboveBelow4_2.json', '06666666666066666666665555555555506666666666000000000000666666666606666666666555555555550666666666600000000000', '0000000000005666666666555555555550666666666500000000000', False, False, 3]


Processing files:  68%|██████▊   | 326/480 [21:47<39:58, 15.57s/file]

['ExtractObjects6_2.json', '99999999999', '2222222222222222', False, False, 3]


Processing files:  68%|██████▊   | 327/480 [21:53<32:28, 12.73s/file]

['TopBottom2D1_1.json', '2222', '4444444444', False, False, 3]


Processing files:  68%|██████▊   | 328/480 [22:17<40:36, 16.03s/file]

['Center4_2.json', '000000000000000800000000077770000000077770006600077770006600000004000000000000000000000022200000000022200000000002220000000000000000000', '0000000000000000000008000007777000000000777700000000077770006600007777000660000000000000000000000000000000002220000000000222000000000022200000000000000000000000000000000', False, False, 3]


Processing files:  69%|██████▊   | 329/480 [22:35<41:44, 16.59s/file]

['Count6_0.json', '8', '4440000', False, False, 3]


Processing files:  69%|██████▉   | 330/480 [22:41<33:32, 13.42s/file]

['ExtractObjects9_0.json', '97775', '77777770777000777777', False, False, 3]


Processing files:  69%|██████▉   | 331/480 [23:07<42:52, 17.27s/file]

['CompleteShape1_0.json', '000000000000000000000000020000000040400000040004000040000040020000000000400000400004000400000040400000000200000000000000', '000000000000000000000000020000000040400000040004000040000040020000000200400000400004000400000040400000000200000000000000', False, True, 3]


Processing files:  69%|██████▉   | 332/480 [23:14<34:48, 14.11s/file]

['FilledNotFilled2_0.json', '00000088800000005555', '000000000000000000', False, False, 3]


Processing files:  69%|██████▉   | 333/480 [23:28<34:38, 14.14s/file]

['TopBottom3D3_1.json', '0555000055500075557770555000055500000000000000000055500005550007555777055500000000000000000', '0000000000000000000007777777000000000000000000000', False, False, 3]


Processing files:  70%|██████▉   | 334/480 [23:34<28:34, 11.74s/file]

['Count7_0.json', '666', '222222', False, False, 3]


Processing files:  70%|██████▉   | 335/480 [23:41<24:52, 10.29s/file]

['TopBottom2D7_0.json', '2', '11', False, False, 3]


Processing files:  70%|███████   | 336/480 [23:48<22:18,  9.29s/file]

['Count8_0.json', '4', '22222', False, False, 3]


Processing files:  70%|███████   | 337/480 [23:57<21:49,  9.16s/file]

['MoveToBoundary1_0.json', '000020000020111120', '000020000020011120000020000020000020', False, False, 3]


Processing files:  70%|███████   | 338/480 [24:38<44:37, 18.86s/file]

['Center4_1.json', '000000000000000000000000111111100000000000000000000000000000006600000000006600000000000000000000444444000000400004000000444444000000000000000', '000000000000000000000000001111111100000000000000000000000000000000000000000000000000000000000000000444444000000400004000000444444000000000000000', False, False, 3]


Processing files:  71%|███████   | 339/480 [25:38<1:13:28, 31.27s/file]

['CompleteShape1_2.json', '0000000000000000000000022202220002220222000000000000022202220002220222000000000000022200000002220000', '0000000000000000000000022202220002220222000000000000022202220002220222000000000000022202220002220222', False, True, 3]


Processing files:  71%|███████   | 340/480 [25:46<56:41, 24.30s/file]  

['CompleteShape4_2.json', '055500', '000000005000000000005000000000000000', False, False, 3]


Processing files:  71%|███████   | 341/480 [26:04<51:18, 22.15s/file]

['ExtendToBoundary1_0.json', '0666666606666666066666660666666606666666066666660000000000000000000000000000000000000000', '000000000666666606666666066666660666666606666666066666660000600000006000000060000000600000006000', False, False, 3]


Processing files:  71%|███████▏  | 342/480 [26:50<1:07:35, 29.39s/file]

['InsideOutside9_2.json', '0000040000002000020002000000200020400200020000002000240002000200040020002000020040004000000000000400', '0000000000002000020002000000200020400200020000002000240002000200040020002000020000000000000000000000', False, True, 3]


Processing files:  71%|███████▏  | 343/480 [27:50<1:28:18, 38.67s/file]

['InsideOutside9_1.json', '2000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000

Processing files:  72%|███████▏  | 344/480 [28:55<1:45:22, 46.49s/file]

['Count7_1.json', '333333333333', '22', False, False, 3]


Processing files:  72%|███████▏  | 345/480 [29:00<1:16:26, 33.97s/file]

['FilledNotFilled2_1.json', '5555555', '0000000000000000000000000000', False, False, 3]


Processing files:  72%|███████▏  | 346/480 [29:01<54:13, 24.28s/file]  

['Count9_2.json', '0000', '0000', True, True, 1]


Processing files:  72%|███████▏  | 347/480 [29:07<41:42, 18.81s/file]

['Count8_1.json', '8', '2', False, True, 3]


Processing files:  72%|███████▎  | 348/480 [29:32<44:54, 20.41s/file]

['FilledNotFilled6_0.json', '00000000000002222222000002000002000002000002000002222222000000000000000000777777700000066666660000006666666000000666666600000000000000000', '000000000000022222220000022222220000022222220000022222220000000000000000000000077770006660070070006060070070006660077770000000000000000000000000', False, False, 3]


Processing files:  73%|███████▎  | 349/480 [29:39<35:52, 16.43s/file]

['ExtendToBoundary5_2.json', '700', '700070007', False, False, 3]


Processing files:  73%|███████▎  | 350/480 [29:55<35:45, 16.50s/file]

['CleanUp2_1.json', '000990000000077777777900777777700007797777000077777970900000000000', '000000000000077777770000777777700007777777000077777770000000000000', False, True, 3]


Processing files:  73%|███████▎  | 351/480 [30:25<44:05, 20.51s/file]

['ExtractObjects10_0.json', '8888777778888', '7777777777774777777777777', False, False, 3]


Processing files:  73%|███████▎  | 352/480 [30:43<42:15, 19.80s/file]

['InsideOutside9_0.json', '222222004040400000222222000000000400000000040000000400000000000000044000004440000000', '222222004040400000222222000000000000000000000000000000000000000000000000000000000000', False, True, 3]


Processing files:  74%|███████▎  | 353/480 [30:53<35:43, 16.88s/file]

['InsideOutside1_0.json', '000033030', '0000', False, False, 3]


Processing files:  74%|███████▍  | 354/480 [31:04<31:35, 15.04s/file]

['ExtractObjects7_2.json', '9999999999', '666666666666666666666666', False, False, 3]


Processing files:  74%|███████▍  | 355/480 [31:12<27:01, 12.97s/file]

['InsideOutside1_1.json', '113', '00000', False, False, 3]


Processing files:  74%|███████▍  | 356/480 [31:20<23:45, 11.50s/file]

['Count1_0.json', '333333', '400000', False, True, 3]


Processing files:  74%|███████▍  | 357/480 [31:40<28:49, 14.06s/file]

['Center1_0.json', '0000000000088800000008080888880888080008000008000800000800080000088888000000000000000000000000000000', '0000000000000000000000000222220000002000200000020202000000200020000002222200000000000000000000000000000000000000000000000', False, False, 3]


Processing files:  75%|███████▍  | 358/480 [31:47<24:12, 11.90s/file]

['TopBottom3D5_2.json', '666636666', '300000030000003000000300000030000003', False, False, 3]


Processing files:  75%|███████▍  | 359/480 [31:53<20:06,  9.97s/file]

['FilledNotFilled3_0.json', '85858585', '33', False, False, 3]


Processing files:  75%|███████▌  | 360/480 [32:11<24:59, 12.50s/file]

['CompleteShape1_1.json', '000000000000000000005555500005500500005555500005555500005555500000000000000000000', '000000000000000000003333300003333300003333300003333300003333300000000000000000000', False, True, 3]


Processing files:  75%|███████▌  | 361/480 [32:24<25:10, 12.69s/file]

['ExtendToBoundary1_1.json', '000000000000', '00000000000000111111101111100011100000100000010000001000000100000010000001000', False, False, 3]


Processing files:  75%|███████▌  | 362/480 [32:40<26:47, 13.62s/file]

['Copy7_0.json', '111111111112221111111111100000000007777777777777777777777777', '111111111112221111111111100000000007777777777722277777777777', False, True, 3]


Processing files:  76%|███████▌  | 363/480 [33:00<30:19, 15.55s/file]

['Copy7_2.json', '877780200028777802000287778020002877780200028777802000288888022222', '888880222228777802777287778027772877780277728777802777288888022222', False, True, 3]


Processing files:  76%|███████▌  | 364/480 [33:06<24:18, 12.57s/file]

['Count1_1.json', '4446666', '444400', False, False, 3]


Processing files:  76%|███████▌  | 365/480 [33:13<20:58, 10.94s/file]

['FilledNotFilled1_0.json', '0', '3', False, True, 3]


Processing files:  76%|███████▋  | 366/480 [33:19<17:52,  9.41s/file]

['FilledNotFilled2_2.json', '111111110', '00000000', False, False, 3]


Processing files:  76%|███████▋  | 367/480 [33:24<15:31,  8.24s/file]

['ExtractObjects9_2.json', '99998', '00800000880000888888808800000080000', False, False, 3]


Processing files:  77%|███████▋  | 368/480 [33:53<26:56, 14.44s/file]

['TopBottom3D2_2.json', '000000000000060600005556555555000606000055565555550006060000000000000000000000000000000000000000000000060600005556555555000606000055565555550006060000000000000000000000000000000000', '000000000000060000005556555555000600000055565555550006000000000000000000000000000000000000', False, False, 3]


Processing files:  77%|███████▋  | 369/480 [34:02<23:21, 12.63s/file]

['ExtractObjects6_0.json', '1111112333', '88885555', False, False, 3]


Processing files:  77%|███████▋  | 370/480 [34:07<19:02, 10.38s/file]

['TopBottom2D1_0.json', '444', '444444', False, False, 3]


Processing files:  77%|███████▋  | 371/480 [34:13<16:44,  9.21s/file]

['CleanUp2_2.json', '', '000000000000000000000000000000000000', False, False, 3]


Processing files:  78%|███████▊  | 372/480 [34:18<14:24,  8.01s/file]

['FilledNotFilled3_1.json', '1', '2222222222', False, False, 3]


Processing files:  78%|███████▊  | 373/480 [34:24<13:07,  7.36s/file]

['FilledNotFilled3_2.json', '1', '4444444444444444444444444444444444444444444444444', False, False, 3]


Processing files:  78%|███████▊  | 374/480 [34:39<16:41,  9.44s/file]

['ExtractObjects7_1.json', '111155333311115530031001553003100155333311115500000000550000', '11113333100130031001300311113333', False, False, 3]


Processing files:  78%|███████▊  | 375/480 [34:53<19:17, 11.03s/file]

['TopBottom3D2_1.json', '000000000003300000999999999999999999003300000003300000003300000003300000000000000', '000000000000000000999999999999999999000000000000000000000000000000000000000000000', False, True, 3]


Processing files:  78%|███████▊  | 376/480 [34:58<15:59,  9.22s/file]

['TopBottom2D1_2.json', '33333', '333303333', False, False, 3]


Processing files:  79%|███████▊  | 377/480 [35:05<14:34,  8.49s/file]

['ExtendToBoundary1_2.json', '01212121212121212', '000000000040000000044444444040000000000000000000000000', False, False, 3]


Processing files:  79%|███████▉  | 378/480 [35:16<15:51,  9.33s/file]

['AboveBelow5_0.json', '00060005', '0006000000060000000600000005000660050006600500065005000550050005', False, False, 3]


Processing files:  79%|███████▉  | 379/480 [35:29<17:24, 10.34s/file]

['Center10_1.json', '055000050000001000000500005500000000000055', '055000050000000000000100000000000005000055000000', False, False, 3]


Processing files:  79%|███████▉  | 380/480 [35:35<14:54,  8.95s/file]

['Count1_2.json', '333333333', '444000', False, False, 3]


Processing files:  79%|███████▉  | 381/480 [35:50<17:43, 10.74s/file]

['AboveBelow5_1.json', '188888188888182888188888188888188888288888188888188288', '000000282200282200282200288200288200288200288200888200888200', False, False, 3]


Processing files:  80%|███████▉  | 382/480 [35:56<15:09,  9.28s/file]

['Count9_0.json', '8888', '00', False, False, 3]


Processing files:  80%|███████▉  | 383/480 [36:05<15:13,  9.41s/file]

['Count9_1.json', '7000700070000000000000000000', '0', False, False, 3]


Processing files:  80%|████████  | 384/480 [36:31<23:00, 14.38s/file]

['FilledNotFilled6_1.json', '0000880000000800800000800008000800000080888888888833333333333333333333033333330000333333000003333000123456708', '0000880000000888800000888888000888888880888888888833333333333000000003030000003000300003000003333000', False, False, 3]


Processing files:  80%|████████  | 385/480 [36:40<20:08, 12.72s/file]

['ExtractObjects7_0.json', '111111111122222888882222200000', '77772222', False, False, 3]


Processing files:  80%|████████  | 386/480 [37:00<23:10, 14.80s/file]

['Order3_1.json', '4444440444444888888044444444444404444444444440888884444444044444444444404444448888880444444444444044444444444408888844444440444444', '000000000000044444404444444444440888888444444044444488888404444444444440444444', False, False, 3]


Processing files:  81%|████████  | 387/480 [37:20<25:17, 16.32s/file]

['Order3_2.json', '2202202202200220220220020002022022002002002002200200220220220220022022022022002202202200200020220220020020020022002002202202202200', '000000000000022022022022002202202200200220220020020022020020002002202202202200', False, False, 3]


Processing files:  81%|████████  | 388/480 [37:30<22:24, 14.62s/file]

['Center10_0.json', '055050', '000000000000055000051500005500000000', False, False, 3]


Processing files:  81%|████████  | 389/480 [39:19<1:04:54, 42.80s/file]

['Center10_2.json', '550000', '000055000005000100050000055000000000', False, False, 3]


Processing files:  81%|████████▏ | 390/480 [39:30<50:00, 33.34s/file]  

['ExtractObjects8_2.json', '55055555555', '050555555555555555', False, False, 3]


Processing files:  81%|████████▏ | 391/480 [39:37<37:51, 25.53s/file]

['ExtractObjects6_1.json', '02222222', '770001000770011100770111110771111111', False, False, 3]


Processing files:  82%|████████▏ | 392/480 [39:47<30:16, 20.64s/file]

['ExtractObjects10_2.json', '55551115555111555511155551915555191555111155511115551111', '111111111191191111111111', False, False, 3]


Processing files:  82%|████████▏ | 393/480 [39:53<23:53, 16.47s/file]

['InsideOutside7_0.json', '82', '1113', False, False, 3]


Processing files:  82%|████████▏ | 394/480 [40:02<20:04, 14.01s/file]

['InsideOutside7_1.json', '53', '11113333', False, False, 3]


Processing files:  82%|████████▏ | 395/480 [40:08<16:33, 11.68s/file]

['InsideOutside7_2.json', '63', '1', False, False, 3]


Processing files:  82%|████████▎ | 396/480 [40:16<14:40, 10.48s/file]

['AboveBelow5_2.json', '999999999999', '111111111111111111111111111111111111111111111111', False, False, 3]


Processing files:  83%|████████▎ | 397/480 [40:23<13:22,  9.66s/file]

['Count10_1.json', '0003000300030003', '00', False, False, 3]


Processing files:  83%|████████▎ | 398/480 [40:37<14:49, 10.85s/file]

['Order3_0.json', '11111111151101101105100110110110051011011111111115', '000000000000000000000011011051015110110110511105105101111011011051', False, False, 3]


Processing files:  83%|████████▎ | 399/480 [40:52<16:11, 11.99s/file]

['TopBottom3D2_0.json', '0101002010100155555530101002010100155555530101002', '0000001000000155555510000001000000155555510000001', False, True, 3]


Processing files:  83%|████████▎ | 400/480 [40:58<13:43, 10.30s/file]

['Count3_1.json', '5', '33', False, False, 3]


Processing files:  84%|████████▎ | 401/480 [41:07<13:04,  9.93s/file]

['ExtendToBoundary4_0.json', '006077', '000000000000006000006000006777006000', False, False, 3]


Processing files:  84%|████████▍ | 402/480 [41:23<15:21, 11.81s/file]

['ExtendToBoundary4_1.json', '000800000000033333000000000000000000000800000000033333000000000000000000', '000800000800033333000000000000000000', False, False, 3]


Processing files:  84%|████████▍ | 403/480 [41:34<14:38, 11.41s/file]

['ExtendToBoundary4_2.json', '280800080800080080008000080080800800', '200008020080002800008000080000800000', False, True, 3]


Processing files:  84%|████████▍ | 404/480 [41:41<13:02, 10.29s/file]

['MoveToBoundary6_0.json', '100000', '000000000000000000000000000001000000', False, False, 3]


Processing files:  84%|████████▍ | 405/480 [41:50<12:10,  9.74s/file]

['MoveToBoundary6_1.json', '410000', '010000000000000000000000000000000000', False, False, 3]


Processing files:  85%|████████▍ | 406/480 [42:02<13:01, 10.56s/file]

['MoveToBoundary6_2.json', '666666000000900000000000000000000000000090', '000000000000000009000000000000000000', False, False, 3]


Processing files:  85%|████████▍ | 407/480 [42:11<12:02,  9.90s/file]

['TopBottom2D9_0.json', '987654', '900550990055099005009900900990990099', False, False, 3]


Processing files:  85%|████████▌ | 408/480 [42:18<10:56,  9.12s/file]

['TopBottom2D9_1.json', '777770', '000555555500500007007777777000000077', False, False, 3]


Processing files:  85%|████████▌ | 409/480 [42:23<09:28,  8.01s/file]

['TopBottom2D9_2.json', '111111', '005058555058000058555558000008888888', False, False, 3]


Processing files:  85%|████████▌ | 410/480 [42:33<09:56,  8.52s/file]

['TopBottom3D8_0.json', '090001090000090000333333090000090001', '090000090000090000393333090000090000', False, True, 3]


Processing files:  86%|████████▌ | 411/480 [42:46<11:10,  9.71s/file]

['TopBottom3D8_1.json', '040000888888040000040000747777040000040000888888040000040000747777', '040000848888040000040000777777040000', False, False, 3]


Processing files:  86%|████████▌ | 412/480 [43:06<14:45, 13.02s/file]

['TopBottom3D8_2.json', '053000350000030000053000050300050030053000350000030000053000050300050030', '050000350000050000053000050300050030', False, False, 3]


Processing files:  86%|████████▌ | 413/480 [43:19<14:24, 12.91s/file]

['CompleteShape2_2.json', '333333', '0000000060000000006066666666660000000060000000006000000000600000000060000000006000000000600000000060', False, False, 3]


Processing files:  86%|████████▋ | 414/480 [43:25<11:47, 10.72s/file]

['InsideOutside1_2.json', '5', '0', False, True, 3]


Processing files:  86%|████████▋ | 415/480 [43:29<09:34,  8.84s/file]

['Count4_1.json', '99999', '0', False, False, 3]


Processing files:  87%|████████▋ | 416/480 [43:44<11:15, 10.56s/file]

['ExtractObjects8_1.json', '111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111', '11111111', False, False, 3]


Processing files:  87%|████████▋ | 417/480 [43:52<10:29,  9.99s/file]

['Count4_2.json', '666666', '000', False, False, 3]


Processing files:  87%|████████▋ | 418/480 [44:00<09:37,  9.32s/file]

['Center4_0.json', '000033030004033', '0000000033333003000300300030030003003333300000000', False, False, 3]


Processing files:  87%|████████▋ | 419/480 [44:05<08:06,  7.97s/file]

['ExtractObjects8_0.json', '66666666', '066660666666066660', False, False, 3]


Processing files:  88%|████████▊ | 420/480 [44:12<07:49,  7.83s/file]

['Count6_1.json', '4', '4444000', False, False, 3]


Processing files:  88%|████████▊ | 421/480 [44:19<07:14,  7.37s/file]

['Count10_0.json', '6000000', '0000', False, False, 3]


Processing files:  88%|████████▊ | 422/480 [44:26<07:01,  7.27s/file]

['ExtractObjects10_1.json', '6666444', '616661661', False, False, 3]


Processing files:  88%|████████▊ | 423/480 [44:31<06:14,  6.57s/file]

['Count3_2.json', '6', '3', False, True, 3]


Processing files:  88%|████████▊ | 424/480 [44:43<07:53,  8.45s/file]

['CompleteShape2_0.json', '000200000002000000020000888888800020000000200000002000000020000888888800020000', '0002000000020000000200000002000000020000000200008888888800020000', False, False, 3]


Processing files:  89%|████████▊ | 425/480 [44:46<06:10,  6.73s/file]

['Order8_1.json', '7164', '7164', True, True, 1]


Processing files:  89%|████████▉ | 426/480 [44:58<07:27,  8.28s/file]

['Order9_0.json', '000000000000000333333333333000333333000333000333333333', '000000000000000333333333333000333333333000333333000333', False, True, 3]


Processing files:  89%|████████▉ | 427/480 [45:08<07:41,  8.71s/file]

['Order9_1.json', '000000000000000000333', '000000000000000000000333333333333333000333333333000333', False, False, 3]


Processing files:  89%|████████▉ | 428/480 [45:25<09:50, 11.36s/file]

['Order9_2.json', '000000333333333000333000333333333333333000333333333333333333333', '000000333333333333333000333333333333000333333333000333', False, False, 3]


Processing files:  89%|████████▉ | 429/480 [45:34<08:51, 10.43s/file]

['TopBottom2D6_2.json', '8', '222222200002200002200002222222', False, False, 3]


Processing files:  90%|████████▉ | 430/480 [45:45<08:50, 10.62s/file]

['CompleteShape2_1.json', '900400000000000400909090000400000000', '000400000400000400999999000400000400', False, True, 3]


Processing files:  90%|████████▉ | 431/480 [45:52<08:00,  9.80s/file]

['Count10_2.json', '7777777', '0', False, False, 3]


Processing files:  90%|█████████ | 432/480 [46:14<10:35, 13.24s/file]

['CleanUp10_2.json', '00000000000200000000002000000000020200000000200000022222222200002000000000000000', '000000020000200000200000020002000000002020000000000200000000002020000000020002000000200000200000000000020000000000002000000000000200000000000020', False, False, 3]


Processing files:  90%|█████████ | 433/480 [46:41<13:36, 17.36s/file]

['CleanUp9_2.json', '000000000000011111000000010002002000210022000000010001002000021111000000000000000000000002021111020000020001000200010001000000010001000002021112', '000000000000011111000000010001000000010001000000010001000000011111000000000000000000000000011111000000010001000000010001000000010001000000011111', False, True, 3]


Processing files:  90%|█████████ | 434/480 [46:50<11:31, 15.04s/file]

['TopBottom2D2_0.json', '8888', '777777777', False, False, 3]


Processing files:  91%|█████████ | 435/480 [46:57<09:24, 12.55s/file]

['TopBottom2D6_0.json', '333', '6666', False, False, 3]


Processing files:  91%|█████████ | 436/480 [47:02<07:33, 10.30s/file]

['Order8_0.json', '831', '87315', False, False, 3]


Processing files:  91%|█████████ | 437/480 [47:08<06:21,  8.88s/file]

['Order8_2.json', '24103', '2438249', False, False, 3]


Processing files:  91%|█████████▏| 438/480 [47:15<05:47,  8.28s/file]

['Count6_2.json', '3', '4400000', False, False, 3]


Processing files:  91%|█████████▏| 439/480 [47:22<05:29,  8.04s/file]

['Count7_2.json', '989999998', '2', False, False, 3]


Processing files:  92%|█████████▏| 440/480 [47:31<05:26,  8.17s/file]

['Count8_2.json', '4', '22', False, False, 3]


Processing files:  92%|█████████▏| 441/480 [47:42<05:59,  9.21s/file]

['Copy6_2.json', '4444444444000004444444444000004444444444', '4444404444444444044444000000000004444404444444444044444000000000004444404444444444044444', False, False, 3]


Processing files:  92%|█████████▏| 442/480 [47:57<06:57, 10.99s/file]

['Copy6_1.json', '070717070000050565050000010131010131070717070000050565050000010131010', '07000707170717070007000000000500050565056505000500000000010001013101310100010', False, False, 3]


Processing files:  92%|█████████▏| 443/480 [48:22<09:13, 14.95s/file]

['Copy6_0.json', '666601111644601331644601331666601111666601111644601331644601331666601111111101331111101331111101331666601111644601331644601331666601111666601111644601331644601331666601111111101331111101331111101331', '666601111644601331644601331666601111000000000666601111644601331644601331666601111', False, False, 3]


Processing files:  92%|█████████▎| 444/480 [48:31<07:54, 13.18s/file]

['TopBottom2D6_1.json', '3330', '30003000300030003333', False, False, 3]


Processing files:  93%|█████████▎| 445/480 [48:50<08:46, 15.05s/file]

['CompleteShape5_1.json', '030330000033044030000003303300000000000003000000003303000000000330003000000000430003000000', '4300000000330340430000033033000000000000430000000033034000000003300034000000003300430000000033000000', False, False, 3]


Processing files:  93%|█████████▎| 446/480 [48:58<07:22, 13.02s/file]

['CompleteShape5_2.json', '44444444', '4300000034440000004400034034004304404400440000000000034004304304400440440000000000340000000044000000', False, False, 3]


Processing files:  93%|█████████▎| 447/480 [49:46<12:55, 23.50s/file]

['CompleteShape6_2.json', '666666666666666666666660000006666000000666600000066660000006666000000666600000066660000006666000000666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666699666666666666666666660000006660000006660000006660000006660000006660000006660000006162366666666666666666660000066600000666000006660000066600000666000006469666666666666666666660000006660000006660000006660000006660000006660000006660000006', '6666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666666', False, False, 3]


Processing files:  93%|█████████▎| 448/480 [50:16<13:32, 25.39s/file]

['CompleteShape7_1.json', '0000000000000000000007700007000000000000000000000000007700000000000000000000000007770007000000000000902', '0000000000000000000007777777000777777700077777770007777777000777777700077777770007777777000000000000', False, False, 3]


Processing files:  94%|█████████▎| 449/480 [50:26<10:45, 20.83s/file]

['TopBottom2D2_2.json', '3333333', '666666666666666666666', False, False, 3]


Processing files:  94%|█████████▍| 450/480 [50:40<09:19, 18.65s/file]

['Copy5_1.json', '99991999902299099990999909999199990229909999099990', '999909999999909999922909229999909999999909999', False, False, 3]


Processing files:  94%|█████████▍| 451/480 [50:56<08:42, 18.01s/file]

['CompleteShape7_2.json', '000000000000000000000000001110000000010000000000001000000000000000000000', '00000000000000000000000000111100001111000011110000111100000000000000000000000000', False, False, 3]


Processing files:  94%|█████████▍| 452/480 [51:06<07:16, 15.58s/file]

['Copy5_0.json', '77707370777010001000', '777737777000777737777000777737777', False, False, 3]


Processing files:  94%|█████████▍| 453/480 [51:23<07:12, 16.02s/file]

['Copy5_2.json', '11116111110011116111110016161161616111116111110016', '111161111000111161111', False, False, 3]


Processing files:  95%|█████████▍| 454/480 [51:34<06:15, 14.44s/file]

['Count5_0.json', '3330', '0000', False, True, 3]


Processing files:  95%|█████████▍| 455/480 [51:43<05:18, 12.74s/file]

['Count5_1.json', '333330', '000', False, False, 3]


Processing files:  95%|█████████▌| 456/480 [51:49<04:22, 10.92s/file]

['Count5_2.json', '111100', '00000000', False, False, 3]


Processing files:  95%|█████████▌| 457/480 [52:09<05:08, 13.42s/file]

['CleanUp10_1.json', '222222000200020200022202000200200202002202200200020202200202', '222222000200000200000200000200000200000200000200000200000200', False, True, 3]


Processing files:  95%|█████████▌| 458/480 [52:20<04:38, 12.68s/file]

['TopBottom3D6_1.json', '55544445', '33333333', False, True, 3]


Processing files:  96%|█████████▌| 459/480 [52:31<04:20, 12.41s/file]

['TopBottom3D6_2.json', '5522112200', '555555555', False, False, 3]


Processing files:  96%|█████████▌| 460/480 [52:43<04:04, 12.23s/file]

['Center7_1.json', '5555500000', '555', False, False, 3]


Processing files:  96%|█████████▌| 461/480 [52:55<03:50, 12.13s/file]

['TopBottom2D2_1.json', '222044', '888', False, False, 3]


Processing files:  96%|█████████▋| 462/480 [53:04<03:21, 11.22s/file]

['TopBottom3D6_0.json', '88888', '4444444444', False, False, 3]


Processing files:  96%|█████████▋| 463/480 [53:32<04:32, 16.04s/file]

['CleanUp10_0.json', '00020020020220022022222200220020202200020020020220022022222200220020202212023142250220210211220032122220040220220050', '002000002000002000222222002000002000', False, False, 3]


Processing files:  97%|█████████▋| 464/480 [53:51<04:34, 17.14s/file]

['CleanUp9_0.json', '000809844740400040470070990799448490000809844740400040470070990799448490000809844740400040470070990799', '000000444440400040400040400040444440', False, False, 3]


Processing files:  97%|█████████▋| 465/480 [54:07<04:12, 16.83s/file]

['CleanUp9_1.json', '343436300003400044300003204003322434333436000003000044000003044003322434355555', '333333300003300003300003300003333333', False, False, 3]


Processing files:  97%|█████████▋| 466/480 [54:22<03:47, 16.23s/file]

['CompleteShape5_0.json', '340340340340430000000330040033000003340340340340430000000330040033000003', '340000330340000330440000340033000043', False, False, 3]


Processing files:  97%|█████████▋| 467/480 [54:34<03:14, 15.00s/file]

['CompleteShape6_0.json', '555500005500005500000000007777007777', '555500555500555500000000007770007770', False, True, 3]


Processing files:  98%|█████████▊| 468/480 [54:46<02:46, 13.91s/file]

['CompleteShape6_1.json', '00000000026000002600088800000022000022', '000000088800088800088800000022000022', False, False, 3]


Processing files:  98%|█████████▊| 469/480 [54:58<02:27, 13.39s/file]

['CompleteShape7_0.json', '00000006666606666006666000666600666690', '000000066666066666066666066666066666', False, False, 3]


Processing files:  98%|█████████▊| 470/480 [55:18<02:33, 15.37s/file]

['HorizontalVertical9_0.json', '6080006080006080006080000099990033336012000601200060120006012000001515151500444460120006012000601200060120000015151515004444', '404000404000404000404000007777007777', False, False, 3]


Processing files:  98%|█████████▊| 471/480 [55:28<02:04, 13.83s/file]

['HorizontalVertical9_1.json', '777777333338333338333338333338077777', '777700400004400004400004400004007777', False, True, 3]


Processing files:  98%|█████████▊| 472/480 [55:39<01:44, 13.07s/file]

['HorizontalVertical9_2.json', '685555680000680000680000066666000000687777', '407777440000440000440000077777000000', False, False, 3]


Processing files:  99%|█████████▊| 473/480 [55:45<01:16, 10.96s/file]

['Center7_2.json', '55555', '55555', True, True, 1]


Processing files:  99%|█████████▉| 474/480 [55:59<01:10, 11.78s/file]

['Copy1_2.json', '222222222222211112211112222222222222', '222222222222222222222222211112211112211112211112222222222222222222222222', False, False, 3]


Processing files:  99%|█████████▉| 475/480 [56:04<00:48,  9.77s/file]

['Center7_0.json', '666', '66', False, False, 3]


Processing files:  99%|█████████▉| 476/480 [56:15<00:40, 10.01s/file]

['Copy1_0.json', '808000000800080000000080808008808000000800080000000080808008', '808808000000080080000000808808', False, False, 3]


Processing files:  99%|█████████▉| 477/480 [56:24<00:29,  9.84s/file]

['Copy1_1.json', '666666666666666666666666666666666666666666', '66666666666666666666', False, False, 3]


Processing files: 100%|█████████▉| 478/480 [56:26<00:14,  7.39s/file]

['Center2_1.json', '6', '6', True, True, 1]


Processing files: 100%|█████████▉| 479/480 [56:29<00:05,  5.99s/file]

['Center2_2.json', '0', '0', True, True, 2]


Processing files: 100%|██████████| 480/480 [56:32<00:00,  7.07s/file]

['Center2_0.json', '5', '5', True, True, 2]


In [28]:
Comparison

[['ExtendToBoundary2_1.json',
  '55555555555555555555555555555',
  '555555555555555555550000003000000000055055555355555550005505000030000005000550500553555500500055050050300050050005505005555555005000550500000000000500055055555555555550005500000000000000000550000000000000000055555555555555555555',
  False,
  False,
  1],
 ['AboveBelow2_2.json',
  '000000000000000055550000000000555500000000000000',
  '00000000000000000000000000000000000000000000000000000000000000000000000000000000033000000000000330000000555505500000005555055000000000000000000000000000000000000000000000000000000000',
  False,
  False,
  1],
 ['ExtendToBoundary2_2.json',
  '0000000000000000999999999999009000000000090090999990000900900',
  '000000000000000000999999999990000999990999990000999990999990000999010009990000999010009990000999010009990000999010009990000999010009990000999999999990000999999999990000999999999990000000000000000',
  False,
  False,
  1],
 ['InsideOutside4_0.json',
  '00000000000000000000000000000088888

In [29]:
df = pd.DataFrame(Comparison, columns=['Task ', 'Raw Model Answer', 'Correct Answer' , 'Model Answer Evaluation' , 'Model Answer Dimension Evaluation', 'Number of Trials' ])

In [30]:
df

Task   \
0    ExtendToBoundary2_1.json   
1          AboveBelow2_2.json   
2    ExtendToBoundary2_2.json   
3       InsideOutside4_0.json   
4       InsideOutside4_1.json   
..                        ...   
475              Copy1_0.json   
476              Copy1_1.json   
477            Center2_1.json   
478            Center2_2.json   
479            Center2_0.json   

                                      Raw Model Answer  \
0                        55555555555555555555555555555   
1     000000000000000055550000000000555500000000000000   
2    0000000000000000999999999999009000000000090090...   
3    0000000000000000000000000000008888888888000080...   
4    0000000000000000000000000000000000000000000000...   
..                                                 ...   
475  8080000008000800000000808080088080000008000800...   
476         666666666666666666666666666666666666666666   
477                                                  6   
478                                                  0   
479                                                  5   

                                        Correct Answer  \
0    5555555555555555555500000030000000000550555553...   
1    0000000000000000000000000000000000000000000000...   
2    0000000000000000009999999999900009999909999900...   
3    0000000000000000000000000000008888888888000080...   
4    0000000000000000000000000000000000000000000000...   
..                                                 ...   
475                     808808000000080080000000808808   
476                               66666666666666666666   
477                                                  6   
478                                                  0   
479                                                  5   

     Model Answer Evaluation  Model Answer Dimension Evaluation  \
0                      False                              False   
1                      False                              False   
2                      False                              False   
3                      False                              False   
4                      False                              False   
..                       ...                                ...   
475                    False                              False   
476                    False                              False   
477                     True                               True   
478                     True                               True   
479                     True                               True   

     Number of Trials  
0                   1  
1                   1  
2                   1  
3                   1  
4                   1  
..                ...  
475                 3  
476                 3  
477                 1  
478                 2  
479                 2  

[480 rows x 6 columns]

In [31]:
excel_filename = 'results_full.xlsx'
df.to_excel(excel_filename, index=False)